### PR TITLE
strip backticks around procedure completion items, correct insertion …

### PR DESCRIPTION
**Pull request by HerrEmil**
also corrects the insertion range of procedures

Before change:
old

After change:
new

Demo: http://strip-backticks.surge.sh/

### CODE REVIEW

File: src/shared/modules/editor/editorDuck.ts [hunk 1]
```diff
@@ -139,11 +161,19 @@ export const populateEditorFromUrlEpic = (some$: any, _store: any) => {
     })
 }
 
+// CypherEditorSupport returns the content attributes of procedures with dots wrapped in backticks, e.g. "`apoc.coll.avg`"
+// This function strips any surrounding backticks before we use the .content value in the completion item provider
```
Comment (OskarDamkjaer) [inline src/shared/modules/editor/editorDuck.ts:165]: 👍

File: src/shared/modules/editor/editorDuck.ts [hunk 2]
```diff
@@ -261,16 +295,14 @@ export const updateEditorSupportSchemaEpic: Epic<Action, any> = (
...
+        labels: state.meta.labels.map(toLabel),
+        relationshipTypes: state.meta.relationshipTypes.map(toRelationshipType),
+        propertyKeys: state.meta.properties.map(toPropertyKey),
         functions: [
           ...cypherFunctions,
           ...state.meta.functions.map(toFunction)
-        ] as string[],
-        procedures: state.meta.procedures.map(toProcedure) as string[]
+        ],
+        procedures: state.meta.procedures.map(toProcedure)
```
Comment (OskarDamkjaer) [inline src/shared/modules/editor/editorDuck.ts:305]: Nice to see these ts improvements!

**Review by OskarDamkjaer — APPROVED** [review]

**Review by oskarhane — COMMENTED** [review]
Backticks enables tokens (not sure what to call them, but labels, prop-types, procedure names etc) to contain characters like `-`.

This would fail: 
```
CREATE (:Kor-v)
```

This would pass: 
```
CREATE (:`Kor-v`)
```

The problem with this change is that nothing will be escaped anymore.
If you create the above label and type a `MATCH` clause to autocomplete, it will autocomplete with `MATCH (n:Kor-v) RETURN n` which would fail. 

So we do want to keep the back-ticks. Is there a way to work around this?
@HerrEmil @OskarDamkjaer

**Comment by OskarDamkjaer** [comment]
@oskarhane We do have the completion-type available, is there an issue with only stripping backticks for procedures?

**Comment by HerrEmil** [comment]
@oskarhane @OskarDamkjaer I added a check for completion type and only strip surrounding backticks on functions and procedures. I've updated the surge link which works with \`Kor-v\` example now

**Review by oskarhane — APPROVED** [review]
Wohoo, that work better!

Can we write some tests to show that this is intended and that we don't break it in the future?

**Comment by HerrEmil** [comment]
Added a few unit tests and plan on merging on green test builds :)

**Review by OskarDamkjaer — APPROVED** [review]
:+1:

**Pull request merged**